Exiled.Events/EventArgs/InteractingShootingTargetEventArgs.cs

```diff
@@ -48,11 +55,48 @@ public InteractingShootingTargetEventArgs(Player player, ShootingTarget shooting
         /// <summary>
         /// Gets the button the player interacted with.
         /// </summary>
-        public ShootingTarget.TargetButton TargetButton { get; }
+        public ShootingTargetButton TargetButton { get; }
+
+        /// <summary>
+        /// Gets or sets the max hp of the target.
+        /// </summary>
+        public int MaxHp
+        {
+            get => maxHp;
+            set
+            {
+                if (!ShootingTarget.IsSynced)
+                    throw new InvalidOperationException("Attempted to set MaxHp while target is in local mode. Set target's IsSynced to true before setting IsAllowed.");
+                maxHp = Mathf.Clamp(value, 1, 256);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the auto reset time of the target.
+        /// </summary>
+        public int AutoResetTime
+        {
+            get => autoResetTime;
+            set
+            {
+                if (!ShootingTarget.IsSynced)
+                    throw new InvalidOperationException("Attempted to set AutoResetTime while target is in local mode. Set target's IsSynced to true before setting IsAllowed.");
+                autoResetTime = Mathf.Clamp(value, 0, 10);
+            }
+        }
 
         /// <summary>
         /// Gets or sets a value indicating whether or not the interaction is allowed.
         /// </summary>
-        public bool IsAllowed { get; set; }
+        public bool IsAllowed
+        {
+            get => isAllowed;
+            set
+            {
+                if (!ShootingTarget.IsSynced)
+                    throw new InvalidOperationException("Attempted to set IsAllowed while target is in local mode. Set target's IsSynced to true before setting IsAllowed.");
```

Comment:
This can be removed. I'm pretty sure you can `ev.IsAllowed = false` without any issues, even if the target is in local mode.

Reply:
I forgot to remove it when I added the AutoResetTime and MaxHp properties. However those are still only allowed to be set while in sync mode or you will mess up the target.

Reply:
Exactly